src/qibocal/fitting/methods.py

```diff
@@ -1073,3 +1076,344 @@ def calibrate_qubit_states_fit(data, x, y, nshots, qubits, degree=True):
+        resonator_type (str): the type of readout resonator ['3D', '2D'].
+        params_fit (dict): Dictionary of parameters for the fit. {"f_rh":f_rh, "g":g, "Ec":Ec, "Ej":Ej}.
+                          freq_rh is the resonator frequency at high power and g in the readout coupling.
+                          If Ec and Ej are missing, the fit is valid in the transmon limit and if they are indicated,
+                          contains the next-order correction.
+
+    Returns:
+        data_fit (Data): Data file with labels and fit parameters.
+
+    """
+
+    if len(params_fit) == 2:
+        # First order approximation params_fit={"f_rh":f_rh, "g":g}
+        quantities = [
+            "curr_sp",
+            "xi",
+            "d",
+            "f_q/f_rh",
+            "g",
+            "f_rh",
+            "f_qs",
+            "f_rs",
+            "f_offset",
+            "C_ii",
+            "qubit",
+        ]
+    else:
+        # Second order approximation params_fit={"f_rh":f_rh, "g":g, "Ec":Ec, "Ej":Ej}
+        quantities = [
+            "curr_sp",
+            "xi",
+            "d",
+            "g",
+            "Ec",
+            "Ej",
+            "f_rh",
+            "f_qs",
+            "f_rs",
+            "f_offset",
+            "C_ii",
+            "qubit",
+        ]
+
+    data_fit = Data(
+        name=f"fits",
+        quantities=quantities,
+    )
+
+    bias_keys = parse(x)
+    frequency_keys = parse(y)
+    for qubit in qubits:
+        qubit_data = (
+            data.df[data.df["qubit"] == qubit]
+            .drop(columns=["qubit", "iteration"])
+            .groupby([frequency_keys[0], bias_keys[0]], as_index=False)
+            .mean()
+        )
+        qubit_data[bias_keys[0]] = (
+            qubit_data[bias_keys[0]].pint.to(bias_keys[1]).pint.magnitude
+        )
+        qubit_data[frequency_keys[0]] = (
+            qubit_data[frequency_keys[0]].pint.to(frequency_keys[1]).pint.magnitude
+        )
+        if resonator_type == "2D":
+            qubit_data["MSR"] = -qubit_data[frequency_keys[0]]
+
+        biases1 = qubit_data[bias_keys[0]]
+        frequencies1 = qubit_data[frequency_keys[0]]
+        msr = qubit_data["MSR"] * 1e6
+        # test
+        import numpy as np
+
+        frequencies, biases = image_to_curve(frequencies1, biases1, msr)
+        x = frequencies1
+        y = biases1
+        frequencies = frequencies  # [4::]
+        biases = biases  # [4::]
+        import matplotlib.pyplot as plt
+
+        min_y = np.min(y)
+        step_y = y[1] - y[0]
+        max_y = np.max(y)
+        leny = int((max_y - min_y) / step_y) + 1
+        lenx = int(len(x) / (leny))
+        max_x = np.max(x)
+        min_x = np.min(x)
+        X = np.linspace(min_x, max_x, lenx)
+        Y = np.linspace(min_y, max_y, leny)
+        X1, Y1 = np.meshgrid(Y, X)
+        M = np.zeros((lenx, leny))
+        for j in range(lenx):
+            M[j, :] = msr[j * leny : j * leny + leny]
```

Comment:
This seems a reshape of msr in a matrix (lenx, leny), I think you could use the `np.reshape` function, for example
```suggestion
        M = np.reshape(msr, (lenx,leny))
```